### PR TITLE
Enable pending block when there is local transaction pending.

### LINKED ISSUE #925: Maintain pending block automatically when we have local pending transactions
When we have local pending transactions (i.e. transactions tagged as being from a local user rather than from the network), then we should kick off a pending block (as we do when mining) and use that for returning 'pending' results from JSONRPC.

### PR DESCRIPTION
**Pull request by tomusdrw**
Closes #925

### CODE REVIEW

**Comment by tomusdrw** [comment]
Build is failing because of some changes on hyper-mio branch. I'm preparing fix on separate branch.

**Comment by tomusdrw** [comment]
Contains #1011

File: miner/src/miner.rs [hunk 1]
```diff
@@ -154,6 +154,26 @@ impl Miner {
 		let mut queue = self.transaction_queue.lock().unwrap();
 		queue.set_gas_limit(gas_limit);
 	}
+
+	/// Returns true if we had to prepare new pending block
+	fn enable_and_prepare_sealing(&self, chain: &BlockChainClient) -> bool {
+		trace!(target: "miner", "enable_and_prepare_sealing: entering");
+		let have_work = self.sealing_work.lock().unwrap().peek_last_ref().is_some();
```
Comment (gavofyork) [inline miner/src/miner.rs]: racey?

File: miner/src/miner.rs [hunk 2]
```diff
@@ -232,29 +252,48 @@ impl MinerService for Miner {
 		Vec<Result<TransactionImportResult, Error>>
 		where T: Fn(&Address) -> AccountDetails {
 		let mut transaction_queue = self.transaction_queue.lock().unwrap();
-		transaction_queue.add_all(transactions, fetch_account)
+		let is_local = false;
```
Comment (gavofyork) [inline miner/src/miner.rs]: what's the point of `is_local`?

Comment (tomusdrw) [inline miner/src/miner.rs]: Just to explain what `false` flag is about - you think it should be inline or changed to enum?

Comment (gavofyork) [inline miner/src/miner.rs]: i typically try to stick to enums over bools. perhaps worth refactoring later?

**Pull request merged**